anx/provider/provider_test.go

```diff
@@ -8,94 +8,109 @@ import (
-
-	t.Run("with incomplete initialization", func(t *testing.T) {
+
+	It("should work with incomplete initialization", func() {
 		p := &anxProvider{
 			// we initialize `providerMetrics` manually because
 			// `anxProvider.setupProviderMetrics()` panics when called a second time
 			providerMetrics: metrics.NewProviderMetrics("anexia", Version),
 		}
 
 		loadbalancer, loadbalancerEnabled := p.LoadBalancer()
-		require.False(t, loadbalancerEnabled)
-		require.Nil(t, loadbalancer)
+		Expect(loadbalancerEnabled).To(BeFalse())
+		Expect(loadbalancer).To(BeNil())
 
 		instancesV2, instancesV2Enabled := p.InstancesV2()
-		require.False(t, instancesV2Enabled)
-		require.Nil(t, instancesV2)
+		Expect(instancesV2Enabled).To(BeFalse())
+		Expect(instancesV2).To(BeNil())
 	})
 
-}
+	Context("Register cloud provider", func() {
+		BeforeEach(func() {
+			Expect(os.Setenv("ANEXIA_TOKEN", "TOKEN")).Error().ToNot(HaveOccurred())
+		})
 
-func TestRegisterCloudProvider(t *testing.T) {
-	require.NoError(t, os.Setenv("ANEXIA_TOKEN", "TOKEN"))
-	t.Cleanup(func() {
-		require.NoError(t, os.Unsetenv("ANEXIA_TOKEN"))
-	})
-	provider, err := cloudprovider.GetCloudProvider("anexia", bytes.NewReader([]byte("customerID: 555")))
-	require.NoError(t, err)
-	require.NotNil(t, provider)
-	anxProvider, ok := provider.(*anxProvider)
-	require.True(t, ok)
-	config := anxProvider.Config()
-	require.Equal(t, "555", config.CustomerID)
-	require.Equal(t, "TOKEN", config.Token)
-}
+		AfterEach(func() {
+			Expect(os.Unsetenv("ANEXIA_TOKEN")).Error().ToNot(HaveOccurred())
+		})
 
-func TestProviderScheme(t *testing.T) {
-	t.Parallel()
-	require.Equal(t, fmt.Sprintf("%s://", configuration.CloudProviderName), configuration.CloudProviderScheme)
-}
+		It("should register cloud provider", func() {
+			provider, err := cloudprovider.GetCloudProvider("anexia", bytes.NewReader([]byte("customerID: 555")))
+			Expect(err).Error().ToNot(HaveOccurred())
+			Expect(provider).ToNot(BeNil())
+
+			anxProvider, ok := provider.(*anxProvider)
+			Expect(ok).To(BeTrue())
+			config := anxProvider.Config()
+			Expect(config.CustomerID).To(Equal("555"))
+			Expect(config.Token).To(Equal("TOKEN"))
+		})
+	})
+})
 
-func TestProviderConfig(t *testing.T) {
-	t.Parallel()
-	provider, err := newAnxProvider(configuration.ProviderConfig{
-		Token:      "5555",
-		CustomerID: "5555",
+var _ = Describe("Config", func() {
+	It("should test provider schema", func() {
+		Expect(fmt.Sprintf("%s://", configuration.CloudProviderName)).To(Equal(configuration.CloudProviderScheme))
 	})
-	require.NoError(t, err)
 
-	config := provider.Config()
+	It("should test provider config", func() {
+		provider, err := newAnxProvider(configuration.ProviderConfig{
+			Token:      "5555",
+			CustomerID: "5555",
+		})
+		Expect(err).Error().ToNot(HaveOccurred())
+
+		config := provider.Config()
+		Expect(config.CustomerID).To(Equal("5555"))
+	})
+})
 
-	require.Equal(t, "5555", config.CustomerID)
+func TestLoadBalancer(t *testing.T) {
```

Comment:
wrong name in this case, maybe use something generic like `TestSuite`

finally found something wrong :D
(which is to say: good job!)

Reply:
:D you got me! I opted for `TestProviderSuite`

Reply:
sorry for the delay, please always feel free to ping me directly when waiting for me